atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/RangeRequest.java

```diff
@@ -183,63 +189,52 @@ public String toString() {
         PtBytes.addIfNotEmpty(helper, "startInclusive", startInclusive);
         PtBytes.addIfNotEmpty(helper, "endExclusive", endExclusive);
         if (columns != null && !columns.isEmpty()) {
-            helper.add("columns", FluentIterable.from(columns).filter(Predicates.notNull()).transform(PtBytes.BYTES_TO_HEX_STRING));
+            helper.add("columns", FluentIterable.from(columns)
+                    .filter(Predicates.notNull())
+                    .transform(PtBytes.BYTES_TO_HEX_STRING));
         }
         helper.add("batchHint", batchHint);
         helper.add("reverse", reverse);
         return helper.toString();
     }
 
     @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        RangeRequest that = (RangeRequest) obj;
+        return reverse == that.reverse
+                && Arrays.equals(startInclusive, that.startInclusive)
+                && Arrays.equals(endExclusive, that.endExclusive)
+                && Objects.equals(columns, that.columns)
+                && Objects.equals(batchHint, that.batchHint);
+    }
+
+    @Override
     public int hashCode() {
         /*
          * Lazily compute and store hashcode since instances are frequently
          * accessed via hash collections, but computation can be expensive, and
          * allow for benign data races.
          */
         if (hashCode == 0) {
-            final int prime = 31;
-            int result = 1;
-            result = prime * result + ((batchHint == null) ? 0 : batchHint.hashCode());
-            result = prime * result + ((columns == null) ? 0 : columns.hashCode());
-            result = prime * result + Arrays.hashCode(endExclusive);
-            result = prime * result + (reverse ? 1231 : 1237);
-            result = prime * result + Arrays.hashCode(startInclusive);
-            hashCode = result;
+            hashCode = Objects.hash(
```

Comment:
This is auto-generated. I just ran IntelliJ's auto-generate equals and hashCode task. It was mainly because equals needed things like braces, etc but I didn't want to manually add them. I did check them to make sure that they still use the same parameters though